docs/vnext/openapi.yaml

```diff
@@ -0,0 +1,325 @@
+openapi: 3.0.3
+info:
+  title: Azure Deployments Extensibility Provider API
+  description: |-
+    This is the OpenAPI 3.0 based specification for Azure Deployments Extensibility Provider API.
+  version: 2.0.0
+externalDocs:
+  description: Find out more about OpenAPI.
+  url: https://www.openapis.org/
+servers:
+  - url: https://{providerApiEndpoint}
+    variables:
+      providerApiEndpoint:
+        default: 'api.example-provider.com'
+        description: Extensibility provider API endpoint.
+paths:
+  /{providerVersion}/resources/reference:
+    post:
+      summary: Create a resource reference
+      description: |
+        Generate a reference to a resource comprising a reference ID, regardless the existence of the resource.

```

Comment:
The reference ID is used in /DELETE, which makes it possible for Deployment Stacks to manage extensible resources. We still need to finalize the design for authentication for the Kubernetes provider (see Anthony's proposal [here](https://gist.github.com/anthony-c-martin/9b7f8e8545c0848f3d8bfee234ba4360#part-1-authentication)), but for MS Graph it's not a problem.